agent/agent.go

```diff
@@ -214,6 +220,35 @@ func (a *Agent) run(ctx context.Context) {
 
 	a.worker.Listen(ctx, reporter)
 
+	updateNode := func() {
+		// skip updating if the registration isn't finished
+		if registered != nil {
+			return
+		}
+		// get the current node description
+		newNodeDescription, err := a.nodeDescriptionWithHostname(ctx, nodeTLSInfo)
```

Comment:
so, `a.nodeDescriptionWithHostname` can return `nil, nil`? Shouldn't we simply `return` inside of the `if err != nil`?